src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -703,9 +703,24 @@ protected virtual ClaimsIdentity CreateClaimsIdentity(JsonWebToken jwtToken, Tok
             return CreateClaimsIdentity(jwtToken, validationParameters, actualIssuer);
         }
 
-        private ClaimsIdentity CreateClaimsIdentity(JsonWebToken jwtToken, TokenValidationParameters validationParameters, string actualIssuer)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="jwtToken"></param>
+        /// <param name="validationParameters"></param>
+        /// <param name="issuer"></param>
+        /// <returns></returns>
+        protected virtual ClaimsIdentity CreateClaimsIdentity(JsonWebToken jwtToken, TokenValidationParameters validationParameters, string issuer)
         {
-            ClaimsIdentity identity = validationParameters.CreateClaimsIdentity(jwtToken, actualIssuer);
+
+            if (jwtToken == null)
+                throw LogHelper.LogArgumentNullException(nameof(jwtToken));
+
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+
+            ClaimsIdentity identity = validationParameters.CreateClaimsIdentity(jwtToken, issuer);
```

Comment:
>issuer) [](http://example.com/codeflow?start=90&length=7)

Will CreateClaimsIdentity handle a null issuer?

Reply:
Seems to be we'd just set the default and log like we do in the other method.

Reply:
> > issuer)
> 
> Will CreateClaimsIdentity handle a null issuer?

CreateClaimsIdentity handles a null issuer, and we have unit tests that takes that into account. We grab a default value in the other method if it is needed.

Reply:
@cesarpayan there has to be an issuer.
If the user has turned off 'validate issuer' and there is no issuer, then the ClaimsIdentity.DefaultIssuer should be used.